apm-sniffer/optional-plugins/customize-enhance-plugin/src/main/java/org/apache/skywalking/apm/plugin/customize/conf/CustomizeConfiguration.java

```diff
@@ -106,14 +108,18 @@ public synchronized void loadForConfiguration() {
      * @throws IOException                  link {@link IOException}
      * @throws SAXException                 link {@link SAXException}
      */
-    private List<Map<String, Object>> resolver() throws ParserConfigurationException, IOException, SAXException {
+    private synchronized List<Map<String, Object>> resolver() throws ParserConfigurationException, IOException, SAXException {
```

Comment:
Are you planning to avoid this in parallel class loader? If so, you should use reentrancelock rather than synchronized